src/lib.rs

```diff
@@ -8,30 +12,66 @@ const RESET: &str = "\u{001B}[0m";
 #[cfg(feature = "colors")]
 const RED: &str = "\u{001B}[31m";
 
+#[cfg(feature = "print-defmt")]
+macro_rules! println {
+    (" ") => {
+        // Do nothing if the string is just a space
+    };
+    ($($arg:tt)*) => {
+        defmt::error!($($arg)*);
+    };
+}
+
+#[cfg(not(feature = "print-defmt"))]
+macro_rules! println {
+    ($($arg:tt)*) => {
+        esp_println::println!($($arg)*);
+    };
+}
+
+fn set_color_code(code: &str)
```

Comment:
`code` might be unused in case of `print-defmt` and produce a warning

Reply:
Fixed